### PR TITLE
Comment prediction added (CommentCode2Seq)

### PR DESCRIPTION
**Pull request by malodetz**

### CODE REVIEW

File: code2seq/data/comment_path_context_dataset.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from typing import Dict, List, Optional
+
+from transformers import RobertaTokenizerFast
+
+from code2seq.data.path_context_dataset import PathContextDataset
+
+tokenizer = RobertaTokenizerFast.from_pretrained("microsoft/codebert-base")
```
Comment (egor-bogomolov) [inline code2seq/data/comment_path_context_dataset.py]: Let's move it to a separate class that will provide tokenizers on demand (without creating them multiple times). Also, let's add option to choose tokenizer in the config file

File: code2seq/data/comment_path_context_dataset.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class CommentPathContextDataset(PathContextDataset):
+    @staticmethod
+    def tokenize_label(raw_label: str, vocab: Dict[str, int], max_parts: Optional[int]) -> List[int]:
+        label_with_spaces = " ".join(raw_label.split(PathContextDataset._separator))
```
Comment (egor-bogomolov) [inline code2seq/data/comment_path_context_dataset.py]: You can replace separator symbol with space rather than split+join

File: code2seq/data/comment_path_context_dataset.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        label_tokens = tokenizer.tokenize(label_with_spaces)
+        if max_parts is None:
+            max_parts = len(label_tokens)
+        label_tokens = [tokenizer.bos_token] + label_tokens[: max_parts - 2] + [tokenizer.eos_token]
+        label_tokens += [tokenizer.pad_token] * (max_parts - len(label_tokens))
```
Comment (egor-bogomolov) [inline code2seq/data/comment_path_context_dataset.py]: The whole function body can be replaced with a single call to tokenizer:
```python
tokenized_snippet = tokenizer(
    raw_label.split(PathContextDataset._separator),
    add_special_tokens=True, 
    padding="max_length" if max_parts else "do_not_pad", 
    max_length=max_parts,
    is_split_into_words=True,
)
```

In order for `is_split_into_words=True` to work, add `add_prefix_space=True` to tokenizer creation.

Comment (egor-bogomolov) [reply code2seq/data/comment_path_context_dataset.py]: The output of tokenizer will be a dict with `input_ids` for tokens and `attention_mask` to point out which tokens come from padding. If you don't need the mask, pass `return_attention_mask=False`.

File: code2seq/model/comment_code2seq.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        self._encoder = self._get_encoder(model_config)
+        output_size = len(tokenizer.get_vocab())
+        decoder_step = LSTMDecoderStep(model_config, output_size, self._pad_idx)
+        self._decoder = Decoder(decoder_step, output_size, tokenizer.eos_token_id, teacher_forcing)
+
+        self._loss = SequenceCrossEntropyLoss(self._pad_idx, reduction="batch-mean")
+
+
+class CommentChrF(Metric):
```
Comment (egor-bogomolov) [inline code2seq/model/comment_code2seq.py]: This class seems out of place here. Let's move metrics to a separate module.

**Review by egor-bogomolov — CHANGES_REQUESTED** [review]
Nice job! Please, fix the tokenizer usage before merging.
